incubator/neo4j/templates/readreplicas-deployment.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: "{{ template "neo4j.replica.fullname" . }}"
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "neo4j.name" . }}
+    component: replica
+spec:
+  replicas: {{ .Values.readReplica.numberOfServers }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "neo4j.name" . }}
+        release: {{ .Release.Name | quote }}
+        component: replica
+    spec:
+      containers:
+      - name: neo4j
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        env:
+          - name: NEO4J_dbms_mode
+            value: READ_REPLICA
+          - name: NUMBER_OF_CORES
+            value: "{{ .Values.core.numberOfServers }}"
+        command:
```

Comment:
I'm just thinking that it could be a good idea to actually add shell scripts as files to the chart and include them using the `Files`object. Might be easier to develop and test them. You can use a decent editor and run Shellcheck across them. I never tried this myself, but maybe you want to make a stab a this.

https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/accessing_files.md

Could also be done in a separate PR, of course.